### PR TITLE
feat: resolve DIDComm messages

### PR DESCRIPTION
**Pull request by maycon-mello**
Methods to decrypt DIDComm messages, will be integrated with the qr-code handler in the app side.

### CODE REVIEW

File: packages/relay-service/tests/relay-service.test.js [hunk 1]
```diff
@@ -93,4 +93,51 @@ describe('Relay service', () => {
...
+          msg: toBase64(didCommMessage),
+        },
+        keyPairDocs: [BOB_KEY_PAIR_DOC],
+      });
+
+      expect(result.payload).toStrictEqual({test: 'test'});
+    });
+
+    it('expect to handle URL', async () => {
+      // TODO: mock axios.get to return didComm message
```
Comment (mike-parkhill) [inline packages/relay-service/tests/relay-service.test.js:124]: when are we coming back for this?

Comment (maycon-mello) [reply packages/relay-service/tests/relay-service.test.js]: sorry, it was work in progress. I've pushed the final commit now

Comment (maycon-mello) [reply packages/relay-service/tests/relay-service.test.js]: added the axios mock, and the ability to resolve messages from URLs

**Review by mike-parkhill — DISMISSED** [review]

**Review by SamHellawell — APPROVED** [review]

**Comment by maycon-mello** [comment]
Pushed new commit to fix linting, + improvement on recipient resolution

File: packages/relay-service/lib/index.js [hunk 2]
```diff
@@ -200,11 +205,20 @@ export async function resolveDidcommMessage({keyPairDocs, message}) {
 
   let result = jwe;
 
-  if (message.to) {
-    const keyPairDoc = keyPairDocs.find(doc => doc.controller === message.to);
+  let recipients = jwe?.recipients?.map(recipient => recipient.header.kid);
+
+  if (recipients) {
```
Comment (mike-parkhill) [inline packages/relay-service/lib/index.js:210]: is it valid to have a message with no recipients?

Comment (maycon-mello) [reply packages/relay-service/lib/index.js:210]: if it's not a didcomm message, yes. Maybe I should rename this variables, its being used to check if didcomm.decrypt must be used or not

Comment (maycon-mello) [reply packages/relay-service/lib/index.js:210]: let me update that

Comment (maycon-mello) [reply packages/relay-service/lib/index.js]: done, i've pushed an update. This case applies for jwt message, which is not encrypted using didcomm, but should be a valid message for us

**Review by mike-parkhill — APPROVED** [review]

**Pull request merged**